framework/h/render_test.go

```diff
@@ -385,21 +386,19 @@ func TestCacheByKeyT2(t *testing.T) {
 
 func TestCacheByKeyConcurrent(t *testing.T) {
 	t.Parallel()
-	renderCount := 0
-	callCount := 0
+	var renderCount, callCount atomic.Uint32
 	cachedItem := CachedPerKey(time.Hour, func() (any, GetElementFunc) {
-		key := "key"
-		if callCount == 3 {
-			key = "key2"
-		}
-		if callCount == 4 {
-			key = "key"
-		}
-		callCount++
-		return key, func() *Element {
-			renderCount++
+		fn := func() *Element {
```

Comment:
If I run the tests in the old way it was written, it doesn't pass anymore, so I think the default cache isn't handling concurrency in the same way anymore

Reply:
thinking maybe the cache interface should be 

```
type Store[K comparable, V any] interface {
	GetOrSet(key K, value V, ttl time.Duration) (V, bool)
	Delete(key K)
	Purge()
	Close()
}
```

thoughts?

Reply:
Looks great. I made one update, I changed WithStore to WithCacheStore. Only because everything is in the 'h.' namespace, so I wanted it to be more specifically named to cache